tools/python/mwm_downloader.py

```diff
@@ -0,0 +1,148 @@
+ROOT = os.path.dirname(os.path.realpath(__file__))
+DEFAULT_DOWNLOAD_DIRECTORY = os.path.join(os.path.dirname(os.path.dirname(ROOT)), 'data')
+MAP_REVISION_REGEXP = re.compile(r'(\d{6})')
+MWM_NAME_REGEXP = re.compile(r'"(\S+\.mwm)"')
+
+logger = logging.getLogger(__name__)
+
+
+def parse_html_by_regexp(url, regexp):
+    def parse_line(line):
+        return regexp.search(line.decode('utf-8'))
+
+    with urllib.request.urlopen(url) as http_response:
+        for line in http_response:
+            parsed_line = parse_line(line)
+            if parsed_line is not None:
+                yield parsed_line.group(1)
+
+
+def get_mwm_versions():
+    return map(int, parse_html_by_regexp(DIRECT_MAP_URL, MAP_REVISION_REGEXP))
+
+
+def get_mwm_names(version_num):
+    version_url = '{base}{version}'.format(base=DIRECT_MAP_URL, version=version_num)
+    return map(urllib.parse.unquote, parse_html_by_regexp(version_url, MWM_NAME_REGEXP))
+
+
+def download_file(url, filename, attempts=3):
+    try:
+        urllib.request.urlretrieve(url, filename=filename)
+    except urllib.error.HTTPError as err:
+        if 400 <= err.code < 500:
+            logger.error('URL \'{url}\' is not found'.format(url=url), exc_info=True)
+            return False
+    except PermissionError:
+        logger.error('Can\'t write file {filename}: permission denied'.format(filename=filename), exc_info=True)
+        return False
+    except urllib.error.URLError:
+        if attempts:
+            time.sleep(3)
+            return download_file(url, filename, attempts-1)
+        logger.error('File {filename} is not loaded'.format(filename=filename), exc_info=True)
+        return False
+    return True
+
+
+def progress_bar(total, progress):
+    """
+    Displays or updates a console progress bar.
+
+    Original source: https://stackoverflow.com/a/15860757/1391441
+    """
+    initial_progress = progress
+    bar_length = 20
+    progress = progress / total
+    block = int(round(bar_length * progress))
+    progress_bar_str = '#' * block + '-' * (bar_length - block)
+    text = '\r[{progress_bar}] {percent:.0f}% {item}/{total}'.format(
+        progress_bar=progress_bar_str,
+        percent=round(progress * 100, 0),
+        item=initial_progress,
+        total=total)
+    sys.stderr.write(text)
+    if progress == 1:
+        sys.stderr.write('\n')
+    sys.stderr.flush()
+
+
+def download_mwm_list(version=None, threads=8, folder=None, mwm_prefix_list=None, quiet=True):
+    versions = list(get_mwm_versions())
+
+    if version is None:
+        # Get latest version by default
+        version = max(versions)
+    elif version not in versions:
+        logger.error('Maps with version {version} is not found in {url}'.format(version=version, url=DIRECT_MAP_URL))
+        return
+
+    mwm_names = get_mwm_names(version)
+    if mwm_prefix_list is not None:
+        mwm_prefixes = '|'.join('^{prefix}'.format(prefix=prefix) for prefix in mwm_prefix_list)
```

Comment:
If I'm not mistaken re.match is doing exactly the prefix search

Reply:
My bad. Fixed!